File: 1539_Kth_Missing_Number.cpp
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        set<int> s(arr.begin(), arr.end()); 
        int index = 1;
        while(true)
        {
            if(s.find(index) == s.end()) k--;
            if(k == 0) return index;
            index++; 
        }
        return -1; 
    }
};

File: 1539_Kth_Missing_Number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        numberSet = set(arr)
        index = 1 
        while True:
            if index not in numberSet:
                 k -= 1
            if k == 0:
                return index 
            index += 1 
        return -1
        
